exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/Foreman.java

```diff
@@ -1191,7 +1195,7 @@ public void failed(final RpcException ex) {
       } else {
         // since this won't be waited on, we can wait to deliver this event once the Foreman is ready
         logger.debug("Failure while sending fragment.  Stopping query.", ex);
-        stateListener.moveToState(QueryState.FAILED, ex);
```

Comment:
I just noticed the comment above, which captures what I meant: we can wait to deliver this event once the Foreman is ready.

Maybe add: this case is only for leaf fragments.